torch/csrc/jit/interpreter.cpp

```diff
@@ -627,21 +628,48 @@ struct CodeImpl {
   }
 
   void emitBailOut(Node* node) {
-    // BailOut node has the `attr::Subgraph` which
-    // contains the original deoptimized version
-    // of a computational graph starting from the
-    // bailout point.
-    // BailOut node's first input is a guarded tensor
-    // the rest are inputs we need to be able to
-    // execute the bailout graph
-    emitLoadInputs(node->inputs().slice(0, 1));
+    auto unoptimized_graph = node->inputs()
+                                 .at(0)
+                                 ->type()
+                                 ->expect<FunctionType>()
+                                 ->function()
+                                 ->graph();
+    // unoptimized graph is at index 0
+    // guarded input is at index 1
+    // the rest of args follow
+    emitLoadInputs(node->inputs().slice(1, 1));
     insertInstruction(GUARD, type_table_.size());
     type_table_.emplace_back(node->outputs().at(0)->type());
     insertInstruction(JF, 0 /* to be patched */);
+
     size_t jf_index = instructions_.size() - 1;
-    emitLoadInputs(node->inputs().slice(1));
+    // note, guaded input is already loaded onto the stack
+    // for GUARD instruction
+    emitLoadInputs(node->inputs().slice(2));
     insertInstruction(TAIL_CALL, function_table_.size());
-    auto func = std::make_shared<Function>("bailout", /*optimize=*/true, node->g(attr::Subgraph), nullptr);
+    TORCH_INTERNAL_ASSERT(node->kind() == prim::BailOut);
+    auto bailout_index = node->i(attr::index);
+    TORCH_INTERNAL_ASSERT(bailout_index >= 0);
+
+    auto build_bailout_graph = [bailout_index,
```

Comment:
Can we add another function call here (it can be defined in the same file). This function is getting big, especially with the closure creation going on here. Something like `createBailoutContinuation(...)`